website/src/pages/ForeignKeysPage/ForeignKeysPage.tsx

```diff
@@ -14,105 +14,140 @@ See the License for the specific language governing permissions and
+import { Button } from "@/components/ui/button";
+import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
+import { Skeleton } from "@/components/ui/skeleton";
+import useApiCall from "@/utils/Hook";
+import { errorApi, formatCompareResult, formatGitRef } from "@/utils/Utils";
+import { PlusCircledIcon } from "@radix-ui/react-icons";
 import ForeignKeysHero from "./components/ForeignKeysHero";
 
-export interface Ref {
-  Name: string;
-  CommitHash: string;
-  Version: {
-    Major: number;
-    Minor: number;
-    Patch: number;
-  };
-  RCnumber: number;
-}
+export const formatTite = (gitRef: string, vitessRefs: VitessRefs): string => {
+  let title = formatGitRef(gitRef);
+  vitessRefs.branches.forEach((branch) => {
+    if (branch.commit_hash.match(gitRef)) {
+      title = branch.name;
+    }
+  });
+  vitessRefs.tags.forEach((branch) => {
+    if (branch.commit_hash.match(gitRef)) {
+      title = branch.name;
+    }
+  });
+  return title;
+};
 
 export default function ForeignKeys() {
   const urlParams = new URLSearchParams(window.location.search);
 
-  const [gitRef, setGitRef] = useState<{ tag: string }>({
-    tag: urlParams.get("tag") || "",
+  const [gitRef, setGitRef] = useState<string>(urlParams.get("tag") || "");
+  const [workload, setWorkload] = useState<{ old: string; new: string }>({
+    old: urlParams.get("oldWorkload") || "",
+    new: urlParams.get("newWorkload") || "",
   });
 
-  const [dataRefs, setDataRefs] = useState<Ref[] | undefined>();
-  const [dataFKs, setDataFKs] = useState<MacrosData[]>([]);
-  const [error, setError] = useState<string | null>(null);
-  const [loading, setLoading] = useState<boolean>(true);
-
-  async function loadRefs() {
-    try {
-      const responseRefs = await fetch(
-        `${import.meta.env.VITE_API_URL}vitess/refs`
-      );
-      const jsonDataRefs = await responseRefs.json();
-      setDataRefs(jsonDataRefs);
-    } catch (error) {
-      setError(errorApi);
-    }
-  }
-
-  async function loadData() {
-    const commits = {
-      tag: dataRefs?.filter((r) => r.Name === gitRef.tag)[0]?.CommitHash || "",
-    };
-
-    setLoading(true);
-    try {
-      const responseFK = await fetch(
-        `${import.meta.env.VITE_API_URL}fk/compare?sha=${commits.tag
-        }`
-      );
-      const jsonDataFKs = await responseFK.json();
-      setDataFKs(jsonDataFKs);
-    } catch (error) {
-      console.error("Error while retrieving data from the API", error);
-      setError(errorApi);
-    } finally {
-      setLoading(false);
-    }
-  }
+  const { data: vitessRefs, isLoading: isVitessRefsLoading } =
+    useApiCall<VitessRefs>(`${import.meta.env.VITE_API_URL}vitess/refs`);
 
-  useEffect(() => {
-    loadRefs();
-  }, []);
+  const {
+    data: data,
+    isLoading: isMacrobenchLoading,
+    error: macrobenchError,
+  } = useApiCall<CompareResult>(
+    `${import.meta.env.VITE_API_URL}fk/compare?sha=${gitRef}`
+  );
```

Comment:
Is it normal that we do a call to the API without the workload arguments set?

Reply:
Wrong discard changes from vscode, thanks !